modules/gapi/src/backends/onnx/gonnxbackend.cpp

```diff
@@ -145,9 +145,39 @@ class ONNXCompiled {
     void run();
 };
 
+static void addCUDAExecutionProvider(Ort::SessionOptions *session_options,
+                                     const cv::gapi::onnx::ep::CUDA &cuda_ep) {
+     OrtCUDAProviderOptions options{};
+     options.device_id = cuda_ep.device_id;
+
+     try {
+        session_options->AppendExecutionProvider_CUDA(options);
+     } catch (const std::exception &e) {
+         std::stringstream ss;
+         ss << "ONNX Backend: Failed to enable CUDA"
+            << " Execution Provider: " << e.what();
+         cv::util::throw_error(std::runtime_error(ss.str()));
+     }
+}
+
+static void addTensorRTExecutionProvider(Ort::SessionOptions *session_options,
+                                         const cv::gapi::onnx::ep::TensorRT &trt_ep) {
+     OrtTensorRTProviderOptions options{};
+     options.device_id = trt_ep.device_id;
```

Comment:
Consider more options:
```
options.trt_max_workspace_size = 2147483648;
options.trt_max_partition_iterations = 10;
options.trt_min_subgraph_size = 5;
options.trt_fp16_enable = 1;
options.trt_engine_cache_enable = 1;
options.trt_engine_cache_path = "ort_tensorrt_cache";
options.trt_dump_subgraphs = 1;  
```

Reply:
Well technically they can be provided via env variable:
https://onnxruntime.ai/docs/execution-providers/TensorRT-ExecutionProvider.html#execution-provider-options

Not clear at this point which of them is important to expose to API